plugin/federation/federation.gotpl

```diff
@@ -217,4 +249,64 @@ func (ec *executionContext) __resolve_entities(ctx context.Context, representati
 		return list
 	}
 }
+
+{{- /* Make sure the required fields are in the given entity representation and return the name of the proper resolver. */ -}}
+
+{{ range $_, $entity := .Entities }}
+	{{- if .Resolvers }}
+
+		func (ec *executionContext) entityResolverNameFor{{$entity.Name}}(ctx context.Context, rep map[string]interface{}) (string, error) {
```

Comment:
From a mental model standpoint, I think it would be a lil better if `ec` was passed in as an argument to `entityResolverNameFor`...  Not a major thing, so if you feel strongly about keeping it as is then no problem.  But in this generated code the execution only really care about being able to call `__resolve__service` and `__resolve_entities` as well as providing an interface for marshaling fields. 

Tho - what do you think about not calling any marshaling functions in the generated `entityResolverNameFor` functions?  We are primarily interested in matching up the shapes of `rep` to determine what resolver to call.  We don't care as much about marshaling the values.  Especially since `rep` values are getting marshaled again before calling the entity resolvers.   That would do two things: 1. Avoid double marshaling for values. 2. Can allow us to not even pass `ec` into these generated function.

Thoughts?

Reply:
The only weird cases would be where an invalid representation in the map[string]interface{} would result in that resolver being chosen, rather than another that had proper reps. Of course, the reverse could happen also, and not be caught. I'll modify it and we can see how it looks.